bleak/backends/winrt/client.py

```diff
@@ -452,27 +471,31 @@ async def unpair(self) -> bool:
             Boolean on whether the unparing was successful.
 
         """
-
-        # New local device information object created since the object from the requester isn't updated
-        device_information = await DeviceInformation.create_from_id_async(
-            self._requester.device_information.id
+        device = await self._create_requester(
+            self._device_info
+            if self._device_info is not None
+            else _address_to_int(self.address)
         )
-        if device_information.pairing.is_paired:
-            unpairing_result = await device_information.pairing.unpair_async()
 
-            if unpairing_result.status not in (
-                DeviceUnpairingResultStatus.UNPAIRED,
-                DeviceUnpairingResultStatus.ALREADY_UNPAIRED,
-            ):
-                raise BleakError(
-                    f"Could not unpair with device: {unpairing_result.status}"
+        if device is not None:
+            try:
+                unpairing_result = (
+                    await device.device_information.pairing.unpair_async()
                 )
-
-            else:
+                if unpairing_result.status not in (
+                    DeviceUnpairingResultStatus.UNPAIRED,
+                    DeviceUnpairingResultStatus.ALREADY_UNPAIRED,
+                ):
+                    raise BleakError(
+                        f"Could not unpair with device: {unpairing_result.status}"
+                    )
                 logger.info("Unpaired with device.")
-                return True
+            finally:
+                device.close()
+        else:
+            logger.debug("No device found to unpair.")
```

Comment:
Should we raise an error instead of silently ignoring unknown devices?

Reply:
I'm really not sure about this. If you only want to remove a device from the system if it is available this is not really an error. But if you expect the device to be available for unpairing this is an error of course.
But raising a ``BleakError`` makes it hard to differentiate between an error caused by the call to WinRT backend and the device not being found. This is a problem if the "device not found" error is no problem for your use case but a "backend error" is. 
If we want to raise here we should introduce a ``DeviceNotFound`` exception to be able to handle both cases in different ways.

Reply:
We have a discussion in #527 about error types. I would say, let's just raise a BleakError for now and consider adding a DeviceNotFound error later in a wider scope.

Reply:
Sounds like a plan.